src/collection/cereal.rs

```diff
@@ -62,6 +68,50 @@ where
     Ok(map)
 }
 
+// Custom deserializer for `Template`. This is useful for deserializing values
+// that are not strings, but should be treated as strings such as numbers,
+// booleans, and nulls.
+impl<'de> Deserialize<'de> for Template {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct TemplateVisitor;
+
+        macro_rules! visit_primitive {
+            ($func:ident, $type:ty) => {
+                fn $func<E>(self, v: $type) -> Result<Self::Value, E>
+                where
+                    E: Error,
+                {
+                    Template::try_from(v.to_string()).map_err(E::custom)
+                }
+            };
+        }
+
+        impl<'de> Visitor<'de> for TemplateVisitor {
+            type Value = Template;
+
+            fn expecting(
+                &self,
+                formatter: &mut std::fmt::Formatter,
+            ) -> std::fmt::Result {
+                formatter.write_str(
+                    "Invalid type, must be a string, number or boolean",
```

Comment:
SORRY one last thing. This results in a clunky error message:

```
invalid type: unit value, expected Invalid type, must be a string, number or boolean
```

It should be just `string, number, or boolean`